app/objects/beatmap.py

```diff
@@ -57,33 +57,49 @@ async def api_get_beatmaps(**params: Any) -> Optional[list[dict[str, Any]]]:
     return None
 
 
+KNOWN_BAD_FILES_MD5 = ["d41d8cd98f00b204e9800998ecf8427e"]  # empty file
+
+
 async def ensure_local_osu_file(
     osu_file_path: Path,
     bmap_id: int,
     bmap_md5: str,
 ) -> bool:
     """Ensure we have the latest .osu file locally,
     downloading it from the osu!api if required."""
-    if (
-        not osu_file_path.exists()
-        or hashlib.md5(osu_file_path.read_bytes()).hexdigest() != bmap_md5
-    ):
-        # need to get the file from the osu!api
-        if app.settings.DEBUG:
-            log(f"Doing osu!api (.osu file) request {bmap_id}", Ansi.LMAGENTA)
-
-        url = f"https://old.ppy.sh/osu/{bmap_id}"
-        async with app.state.services.http_client.get(url) as resp:
-            if resp.status != 200:
-                if 400 <= resp.status < 500:
-                    # client error, report this to cmyui
-                    stacktrace = app.utils.get_appropriate_stacktrace()
-                    await app.state.services.log_strange_occurrence(stacktrace)
-                return False
 
-            osu_file_path.write_bytes(await resp.read())
-
-    return True
+    exists = osu_file_path.exists()
+    # some callers have type Any | str on "bmap_md5" so I assume it's optional, thus May be None.
+    # I use a magic string here to avoid letting optional md5 slip through.
+    file_md5 = (
+        hashlib.md5(osu_file_path.read_bytes()).hexdigest()
+        if exists
+        else "MAGIC_STRING__NOT_EXISTS"
+    )
+    if file_md5 == bmap_md5:
+        return True
+    # need to get the file from the osu!api
+    if app.settings.DEBUG:
+        log(f"Doing osu!api (.osu file) request {bmap_id}", Ansi.LMAGENTA)
+
+    url = f"https://old.ppy.sh/osu/{bmap_id}"
+    async with app.state.services.http_client.get(url) as resp:
+        if resp.status != 200:
+            if 400 <= resp.status < 500:
+                # client error, report this to cmyui
+                stacktrace = app.utils.get_appropriate_stacktrace()
+                await app.state.services.log_strange_occurrence(stacktrace)
+            return False
+        b_beatmap = await resp.read()
+        bytes_md5 = hashlib.md5(b_beatmap).hexdigest()
+        if bytes_md5 in KNOWN_BAD_FILES_MD5:
+            return False
+        # at least this' not a known bad file
+        osu_file_path.write_bytes(b_beatmap)
+        # some callers have type Any | str on "bmap_md5" so I assume it's optional
```

Comment:
please avoid the use of random unnecessary comments

Reply:
this is removed due to no longer compare against md5s